ansible-ipi-install/roles/node-prep/tasks/01_sub_man_register.yml

```diff
@@ -0,0 +1,27 @@
+---
+- name: Check if rhsm.conf.kat-backup exists
+  stat:
+    path: /etc/rhsm/rhsm.conf.kat-backup
+  register: rhsmfile
+
+- name: "Check if any repos are enabled currently"
+  shell: 'subscription-manager repos --list-enabled | sed -ne "s/^Repo ID:[^a-zA-Z0-9]*\(.*\)/\1/p"'
```

Comment:
I'm not sure if this is needed?

Reply:
So the redhat_subscription module, if a playbook is run multiple times basically messes up the repos. In order to combat this, I do a simple check to see if any repos are enabled, if they are, then there is a likely possibility that you don't need to go through the registration process again. Is it perfect and catch every use case? Probably not. I'm open to cleaner ways but for our purposes this catches most use cases.